### PR TITLE
Draw Lines and SP Bars Closes #817

### LINKED ISSUE #817: Implement SP bars
Will need a line renderer

### PR DESCRIPTION
**Pull request by Zychronix**
Draw lines and SP bars updated to OpenGL.

### CODE REVIEW

File: src/main/java/legend/game/combat/ui/BattleHud.java [hunk 1]
```diff
@@ -737,23 +737,23 @@ private void drawUiElements() {
...
+                  .monochrome(2, 229.0f / 255.0f)
+                  .monochrome(3, 133.0f / 255.0f)
+                  .size(1, 1)
+                  .build();
+              }
 
-              cmd
-                .rgb(2, spBarColours[3], spBarColours[4], spBarColours[5])
-                .rgb(3, spBarColours[3], spBarColours[4], spBarColours[5]);
+              final MV transforms = new MV();
```
Comment (LordMonoxide) [inline src/main/java/legend/game/combat/ui/BattleHud.java:752]: We don't want to instantiate anything in the game loop, make this a property on the object like spBars

File: src/main/java/legend/game/combat/ui/BattleHud.java [hunk 2]
```diff
@@ -737,23 +737,23 @@ private void drawUiElements() {
...
+              transforms.transfer.set(GPU.getOffsetX() + left, GPU.getOffsetY() + top, 31.0f);
+              transforms.scaling(right - left, bottom - top, 31.0f);
```
Comment (LordMonoxide) [inline src/main/java/legend/game/combat/ui/BattleHud.java:754]: Z scale should be 1.0f, you're scaling the Z axis by 31x

File: src/main/java/legend/game/combat/ui/BattleHud.java [hunk 3]
```diff
@@ -3014,17 +3014,17 @@ private PlayerBattleEntity setActiveCharacterSpell(final int spellId) {
...
   @Method(0x800f9ee8L)
   private void drawLine(final int x1, final int y1, final int x2, final int y2, final int r, final int g, final int b, final boolean translucent) {
-    final GpuCommandLine cmd = new GpuCommandLine()
-      .rgb(0, r, g, b)
-      .rgb(1, r, g, b)
-      .pos(0, x1, y1)
-      .pos(1, x2, y2);
+    final MV transforms = new MV();
+    transforms.transfer.set(GPU.getOffsetX() + x1, GPU.getOffsetY() + y1, 31.0f);
+    transforms.scaling(x2 - x1 + 1, y2 - y1 + 1, 31.0f);
```
Comment (LordMonoxide) [inline src/main/java/legend/game/combat/ui/BattleHud.java:3019]: Same here re: Z scale, just check to make sure it still looks as you expect it to after fixing

**Review by LordMonoxide — CHANGES_REQUESTED** [review]

File: src/main/java/legend/game/combat/ui/BattleHud.java (hunk 1 above)
Comment (Zychronix) [reply src/main/java/legend/game/combat/ui/BattleHud.java]: Made spBarTransforms final below spBars variable.

File: src/main/java/legend/game/combat/ui/BattleHud.java (hunk 2 above)
Comment (Zychronix) [reply src/main/java/legend/game/combat/ui/BattleHud.java]: Changed Z to 1.

File: src/main/java/legend/game/combat/ui/BattleHud.java (hunk 3 above)
Comment (Zychronix) [reply src/main/java/legend/game/combat/ui/BattleHud.java:3019]: Changed Z to 1.

File: src/main/java/legend/game/combat/ui/BattleHud.java [hunk 4]
```diff
@@ -737,23 +738,22 @@ private void drawUiElements() {
...
+                  .monochrome(2, 229.0f / 255.0f)
+                  .monochrome(3, 133.0f / 255.0f)
+                  .size(1, 1)
+                  .build();
+              }
 
-              cmd
-                .rgb(2, spBarColours[3], spBarColours[4], spBarColours[5])
-                .rgb(3, spBarColours[3], spBarColours[4], spBarColours[5]);
+              spBarTransforms.transfer.set(GPU.getOffsetX() + left, GPU.getOffsetY() + top, 1.0f);
```
Comment (LordMonoxide) [inline src/main/java/legend/game/combat/ui/BattleHud.java:753]: The Z translation should still be 31.0f, we just don't want to be scaling the Z axis by 31x on top of that

File: src/main/java/legend/game/combat/ui/BattleHud.java (hunk 3 above)
Comment (LordMonoxide) [inline src/main/java/legend/game/combat/ui/BattleHud.java:3019]: Missed this one

**Review by LordMonoxide — CHANGES_REQUESTED** [review]

**Pull request merged**